src/Dropdown/index.jsx

```diff
@@ -1,203 +1,125 @@
+  componentDidUpdate = (prevProps, prevState) => {
+    if (prevState.open && this.toggleElem) {
       this.toggleElem.focus();
     }
   }
 
-  addEvents() {
-    document.addEventListener('click', this.handleDocumentClick, true);
+  componentWillUnmount = () => {
+    document.removeEventListener('click', this.handleDocumentClick, true);
   }
 
-  removeEvents() {
-    document.removeEventListener('click', this.handleDocumentClick, true);
+  getFocusableElements = () => {
+    const selector = 'button:not([disabled]), [href]:not([disabled]), input:not([disabled]), select:not([disabled]), textarea:not([disabled]), [tabindex]:not([tabindex="-1"]):not([disabled])';
+    return this.dropdownItemRef.current.querySelectorAll(selector);
+  }
+
+  focusFirst = () => {
+    this.getFocusableElements()[0].focus();
+  }
+
+  focusNext = () => {
+    const allFocusableElements = Array.from(this.getFocusableElements());
+    const activeIndex = allFocusableElements.indexOf(document.activeElement);
+    const nextIndex = (activeIndex + 1) % allFocusableElements.length;
+    allFocusableElements[nextIndex].focus();
   }
 
-  handleDocumentClick(e) {
+  focusPrevious = () => {
+    const allFocusableElements = Array.from(this.getFocusableElements());
+    const activeIndex = allFocusableElements.indexOf(document.activeElement);
+    const previousIndex =
+      ((activeIndex - 1) + allFocusableElements.length) % allFocusableElements.length;
+    allFocusableElements[previousIndex].focus();
+  }
+
+  handleDocumentClick = (e) => {
     if (this.container && this.container.contains(e.target) && this.container !== e.target) {
       return;
     }
-    this.toggle();
-  }
-
-  handleMenuKeyDown(e) {
-    e.preventDefault();
-    if (Dropdown.isTriggerKey('CLOSE_MENU', e.key)) {
+    if (this.state.open) {
       this.toggle();
-    } else if (Dropdown.isTriggerKey('SELECT_MENU_ITEM', e.key)) {
-      e.target.click();
-      this.setState({
-        open: false,
-      });
-    } else if (Dropdown.isTriggerKey('NAVIGATE_DOWN', e.key)) {
-      this.setState({
-        focusIndex: (this.state.focusIndex + 1) % this.props.menuItems.length,
-      });
-    } else if (Dropdown.isTriggerKey('NAVIGATE_UP', e.key)) {
-      this.setState({
-        focusIndex: ((this.state.focusIndex - 1) + this.props.menuItems.length) %
-                    this.props.menuItems.length,
-      });
     }
   }
 
-  handleToggleKeyDown(e) {
-    if (!this.state.open && Dropdown.isTriggerKey('OPEN_MENU', e.key)) {
-      this.toggle();
-    } else if (this.state.open && Dropdown.isTriggerKey('CLOSE_MENU', e.key)) {
+  handleMenuKeyDown = (e) => {
+    e.preventDefault();
+    if (e.key === 'ArrowUp') {
+      this.focusPrevious();
+    } else if (e.key === 'ArrowDown' || 'Tab') {
+      this.focusNext();
+    }
+    if (e.key === 'Escape') {
       this.toggle();
     }
   }
 
-  toggle() {
+  open = () => {
+    setTimeout(() => { this.focusFirst(); }, 1);
```

Comment:
It may be worth moving this call to `focusFirst()` inside `componentDidUpdate`. This will remove the need for setTimeout I think